### PR TITLE
Automatically convert TOKXXXX to the TOK enum

### PR DESCRIPTION
**Pull request by wilzbach**
I found one more.

```bash
replacements=(
"TOKreserved reserved"
"TOKlparen leftParentheses"
"TOKrparen rightParentheses"
"TOKlbracket leftBracket"
"TOKrbracket rightBracket"
"TOKlcurly leftCurly"
"TOKrcurly rightCurly"
"TOKcolon colon"
"TOKneg negate"
"TOKsemicolon semicolon"
"TOKdotdotdot dotDotDot"
"TOKeof endOfFile"
"TOKcast cast_"
"TOKnull null_"
"TOKassert assert_"
"TOKtrue true_"
"TOKfalse false_"
"TOKcall call"
"TOKaddress address"
"TOKdelete delete_"
"TOKstar star"
"TOKsymoff symbolOffset"
"TOKvar variable"
"TOKdotvar dotVariable"
"TOKdotid dotIdentifier"
"TOKdotti dotTemplateInstance"
"TOKdottype dotType"
"TOKslice slice"
"TOKarraylength arrayLength"
"TOKversion version_"
"TOKdollar dollar"
"TOKtemplate template_"
"TOKdottd dotTemplateDeclaration"
"TOKdeclaration declaration"
"TOKpragma pragma_"
"TOKdsymbol dSymbol"
"TOKuadd uadd"
"TOKremove remove"
"TOKnewanonclass newAnonymousClass"
"TOKcomment comment"
"TOKarrayliteral arrayLiteral"
"TOKassocarrayliteral assocArrayLiteral"
"TOKstructliteral structLiteral"
"TOKclassreference classReference"
"TOKthrownexception thrownException"
"TOKdelegateptr delegatePointer"
"TOKdelegatefuncptr delegateFunctionPointer"
"TOKlt lessThan"
"TOKgt greaterThan"
"TOKge greaterOrEqual"
"TOKequal equal"
"TOKnotequal notEqual"
"TOKidentity identity"
"TOKnotidentity notIdentity"
"TOKindex index"
"TOKis is_"
"TOKunord unord"
"TOKlg lg"
"TOKleg leg"
"TOKule ule"
"TOKul ul"
"TOKuge uge"
"TOKug ug"
"TOKue ue"
"TOKshlass leftShiftAssign"
"TOKshrass rightShiftAssign"
"TOKushrass unsignedRightShiftAssign"
"TOKcatass concatenateAssign"
"TOKcatelemass concatenateElemAssign"
"TOKcatdcharass concatenateDcharAssign"
"TOKaddass addAssign"
"TOKminass minAssign"
"TOKmulass mulAssign"
"TOKdivass divAssign"
"TOKmodass modAssign"
"TOKandass andAssign"
"TOKorass orAssign"
"TOKxorass xorAssign"
"TOKassign assign"
"TOKtilde tilde"
"TOKplusplus plusPlus"
"TOKminusminus minusMinus"
"TOKconstruct construct"
"TOKblit blit"
"TOKarrow arrow"
"TOKcomma comma"
"TOKquestion question"
"TOKandand andAnd"
"TOKoror orOr"
"TOKpreplusplus prePlusPlus"
"TOKpreminusminus preMinusMinus"
"TOKint32v int32Literal"
"TOKuns32v uns32Literal"
"TOKint64v int64Literal"
"TOKuns64v uns64Literal"
"TOKint128v int128Literal"
"TOKuns128v uns128Literal"
"TOKfloat32v float32Literal"
"TOKfloat64v float64Literal"
"TOKfloat80v float80Literal"
"TOKimaginary32v imaginary32Literal"
"TOKimaginary64v imaginary64Literal"
"TOKimaginary80v imaginary80Literal"
"TOKcharv charLiteral"
"TOKwcharv wcharLiteral"
"TOKdcharv dcharLiteral"
"TOKidentifier identifier"
"TOKstring string_"
"TOKxstring hexadecimalString"
"TOKthis this_"
"TOKsuper super_"
"TOKhalt halt"
"TOKtuple tuple"
"TOKerror error"
"TOKint8 int8"
"TOKuns8 uns8"
"TOKint16 int16"
"TOKuns16 uns16"
"TOKint32 int32"
"TOKuns32 uns32"
"TOKint64 int64"
"TOKuns64 uns64"
"TOKint128 int128"
"TOKuns128 uns128"
"TOKfloat32 float32"
"TOKfloat64 float64"
"TOKfloat80 float80"
"TOKimaginary32 imaginary32"
"TOKimaginary64 imaginary64"
"TOKimaginary80 imaginary80"
"TOKcomplex32 complex32"
"TOKcomplex64 complex64"
"TOKcomplex80 complex80"
"TOKchar char_"
"TOKwchar wchar_"
"TOKdchar dchar_"
"TOKbool bool_"
"TOKstruct struct_"
"TOKclass class_"
"TOKinterface interface_"
"TOKunion union_"
"TOKenum enum_"
"TOKimport import_"
"TOKalias alias_"
"TOKoverride override_"
"TOKdelegate delegate_"
"TOKfunction function_"
"TOKmixin mixin_"
"TOKalign align_"
"TOKextern extern_"
"TOKprivate private_"
"TOKprotected protected_"
"TOKpublic public_"
"TOKexport export_"
"TOKstatic static_"
"TOKconst const_"
"TOKabstract abstract_"
"TOKdebug debug_"
"TOKdeprecated deprecated_"
"TOKout out_"
"TOKinout inout_"
"TOKlazy lazy_"
"TOKauto auto_"
"TOKpackage package_"
"TOKmanifest manifest"
"TOKimmutable immutable_"
"TOKif if_"
"TOKelse else_"
"TOKwhile while_"
"TOKswitch switch_"
"TOKcase case_"
"TOKdefault default_"
"TOKbreak break_"
"TOKcontinue continue_"
"TOKwith with_"
"TOKsynchronized synchronized_"
"TOKreturn return_"
"TOKgoto goto_"
"TOKtry try_"
"TOKfinally finally_"
"TOKasm asm_"
"TOKforeach_reverse foreach_reverse_"
"TOKscope scope_"
"TOKon_scope_exit onScopeExit"
"TOKon_scope_failure onScopeFailure"
"TOKon_scope_success onScopeSuccess"
"TOKinvariant invariant_"
"TOKunittest unittest_"
"TOKargTypes argumentTypes"
"TOKref ref_"
"TOKmacro macro_"
"TOKparameters parameters"
"TOKtraits traits"
"TOKoverloadset overloadSet"
"TOKpure pure_"
"TOKgshared gshared"
"TOKline line"
"TOKfilefullpath fileFullPath"
"TOKmodulestring moduleString"
"TOKfuncstring functionString"
"TOKprettyfunc prettyFunction"
"TOKshared shared_"
"TOKat at"
"TOKpowass powAssign"
"TOKgoesto goesTo"
"TOKvector vector"
"TOKpound pound"
"TOKinterval interval"
"TOKvoidexp voidExpression"
"TOKcantexp cantExpression"
"TOKMAX max_"
"TOKwild inout_"
"TOKnothrow nothrow_"
"TOKcatch catch_"
"TOKtypeof typeof_"
"TOKtypeid typeid_"
"TOKforeach foreach_"
"TOKfor for_"
"TOKfinal final_"
"TOKcat concatenate"
"TOKnot not"
"TOKtype type"
"TOKthrow throw_"
"TOKin in_"
"TOKvoid void_"
"TOKmodule module_"
"TOKnew new_"
"TOKmod mod"
"TOKarray array"
"TOKfile file"
"TOKpow pow"
"TOKor or"
"TOKxor xor"
"TOKmin min"
"TOKmul mul"
"TOKdiv div"
"TOKand and"
"TOKshl leftShift"
"TOKdot dot"
"TOKdo do_"
"TOKadd add"
"TOKshr rightShift"
"TOKushr unsignedRightShift"
"TOKle lessOrEqual"
)
shopt -s globstar
for r in "${replacements[@]}" ; do
    w=($r)
    sed "s/${w[0]}/TOK.${w[1]}/g" -i **/*.d
done
```

### CODE REVIEW

File: src/dmd/tokens.d [hunk 1]
```diff
@@ -24,539 +24,292 @@ import dmd.utf;
...
-alias TOKgoesto = TOK.TOKgoesto;
-alias TOKvector = TOK.TOKvector;
-alias TOKpound = TOK.TOKpound;
-alias TOKinterval = TOK.TOKinterval;
-alias TOKvoidexp = TOK.TOKvoidexp;
-alias TOKcantexp = TOK.TOKcantexp;
-
-alias TOKMAX = TOK.TOKMAX;
-
-enum TOKwild = TOKinout;
```
Comment (wilzbach) [inline src/dmd/tokens.d]: `TOKwild` was replaced with `TOKinout`, but I'm not 100% sure here. What was the reason for the `TOKwild` enum?

**Comment by JinShil** [comment]
Style suggestions:  `arrayLength`, `newAnonClass`, `onScopeFailure` etc...

I don't recommend `forEach`.

**Comment by wilzbach** [comment]
Like this?

```diff
diff --git a/src/dmd/tokens.d b/src/dmd/tokens.d
index e85c5bc8e..e1be563a6 100644
--- a/src/dmd/tokens.d
+++ b/src/dmd/tokens.d
@@ -58,7 +58,7 @@ enum TOK : int
     dotti,
     dottype,
     slice,
-    arraylength,
+    arrayLength,
     version_,
     module_,
     dollar,
@@ -71,15 +71,15 @@ enum TOK : int
     typeid_,
     uadd,
     remove,
-    newanonclass,
+    newAnonClass,
     comment,
-    arrayliteral,
-    assocarrayliteral,
-    structliteral,
-    classreference,
-    thrownexception,
-    delegateptr,
-    delegatefuncptr,
+    arrayLiteral,
+    assocArrayLiteral,
+    structLiteral,
+    classReference,
+    thrownException,
+    delegatePtr,
+    delegateFuncptr,
 
     // 54
     // Operators
@@ -88,9 +88,9 @@ enum TOK : int
     le,
     ge,
     equal,
-    notequal,
+    notEqual,
     identity,
-    notidentity,
+    notIdentity,
     index,
     is_,
 
@@ -265,9 +265,9 @@ enum TOK : int
     foreach_,
     foreach_reverse_,
     scope_,
-    on_scope_exit,
-    on_scope_failure,
-    on_scope_success,
+    onScopeExit,
+    onScopeFailure,
+    onScopeSuccess,
 
     // 215
     // Contracts
@@ -284,16 +284,16 @@ enum TOK : int
     // 221
     parameters,
     traits,
-    overloadset,
+    overloadSet,
     pure_,
     nothrow_,
     gshared,
     line,
     file,
-    filefullpath,
-    modulestring,
-    funcstring,
-    prettyfunc,
+    fileFullPath,
+    moduleString,
+    funcString,
+    prettyFunc,
     shared_,
     at,
     pow,
```

File: src/dmd/tokens.d [hunk 2]
```diff
@@ -24,539 +24,292 @@ import dmd.utf;
...
     // 239
-    TOKinterval,
-    TOKvoidexp,
-    TOKcantexp,
+    interval,
+    voidexp,
+    cantexp,
 
-    TOKMAX,
+    MAX,
```
Comment (wilzbach) [inline src/dmd/tokens.d]: @JinShil should I lowercase this?

File: src/dmd/tokens.d [hunk 3]
```diff
@@ -24,539 +24,292 @@ import dmd.utf;
...
+    vector,
+    pound,
 
     // 239
-    TOKinterval,
-    TOKvoidexp,
-    TOKcantexp,
+    interval,
+    voidexp,
+    cantexp,
```
Comment (wilzbach) [inline src/dmd/tokens.d]: @JinShil I wasn't sure here. `voidExp`? `cantExp`?

File: src/dmd/tokens.d (hunk 2 above)
Comment (ibuclaw) [reply src/dmd/tokens.d]: .max would be a builtin property

Comment (wilzbach) [reply src/dmd/tokens.d]: `max_`?

File: src/dmd/tokens.d (hunk 3 above)
Comment (JinShil) [reply src/dmd/tokens.d]: I suggest `voidExp` and `cantExp`.

**Comment by JinShil** [comment]
> diff --git a/src/dmd/tokens.d b/src/dmd/tokens.d

That diff looks good, but I suggest the following change:  `dotType`

With `dotti`, the "ti" means "Template Instance".  I couldn't find anything about abbreviations in the style guide.  I think it would be `dotTI`.  What does phobos do for abbreviations?

**Comment by jacob-carlborg** [comment]
> I found one more.

Another small one 😉.

**Comment by JinShil** [comment]
Here's my interpretation

```diff
@@ -1,14 +1,14 @@
 "TOKreserved reserved"
-"TOKlparen lparen"
-"TOKrparen rparen"
-"TOKlbracket lbracket"
-"TOKrbracket rbracket"
-"TOKlcurly lcurly"
-"TOKrcurly rcurly"
+"TOKlparen lParen"
+"TOKrparen rParen"
+"TOKlbracket lBracket"
+"TOKrbracket rBracket"
+"TOKlcurly lCurly"
+"TOKrcurly rCurly"
 "TOKcolon colon"
 "TOKneg neg"
 "TOKsemicolon semicolon"
-"TOKdotdotdot dotdotdot"
+"TOKdotdotdot dotDotDot"
 "TOKeof eof"
 "TOKcast cast_"
 "TOKnull null_"
@@ -19,21 +19,21 @@
 "TOKaddress address"
 "TOKdelete delete_"
 "TOKstar star"
-"TOKsymoff symoff"
+"TOKsymoff symOff"
 "TOKvar var"
-"TOKdotvar dotvar"
-"TOKdotid dotid"
-"TOKdotti dotti"
-"TOKdottype dottype"
+"TOKdotvar dotVar"
+"TOKdotid dotID"
+"TOKdotti dotTI"
+"TOKdottype dotType"
 "TOKslice slice"
 "TOKarraylength arrayLength"
 "TOKversion version_"
 "TOKdollar dollar"
 "TOKtemplate template_"
-"TOKdottd dottd"
+"TOKdottd dotTD"
 "TOKdeclaration declaration"
 "TOKpragma pragma_"
-"TOKdsymbol dsymbol"
+"TOKdsymbol dSymbol"
 "TOKuadd uadd"
 "TOKremove remove"
 "TOKnewanonclass newAnonClass"
@@ -44,7 +44,7 @@
 "TOKclassreference classReference"
 "TOKthrownexception thrownException"
 "TOKdelegateptr delegatePtr"
-"TOKdelegatefuncptr delegateFuncptr"
+"TOKdelegatefuncptr delegateFuncPtr"
 "TOKlt lt"
 "TOKgt gt"
 "TOKle le"
@@ -65,42 +65,42 @@
 "TOKue ue"
 "TOKshl shl"
 "TOKshr shr"
-"TOKshlass shlass"
-"TOKshrass shrass"
+"TOKshlass shlAss"
+"TOKshrass shrAss"
 "TOKushr ushr"
-"TOKushrass ushrass"
-"TOKcatass catass"
-"TOKcatelemass catelemass"
-"TOKcatdcharass catdcharass"
+"TOKushrass ushrAss"
+"TOKcatass catAss"
+"TOKcatelemass catElemAss"
+"TOKcatdcharass catdcharAss"
 "TOKadd add"
 "TOKmin min"
-"TOKaddass addass"
-"TOKminass minass"
+"TOKaddass addAss"
+"TOKminass minAss"
 "TOKmul mul"
 "TOKdiv div"
-"TOKmulass mulass"
-"TOKdivass divass"
-"TOKmodass modass"
+"TOKmulass mulAss"
+"TOKdivass divAss"
+"TOKmodass modAss"
 "TOKand and"
 "TOKor or"
 "TOKxor xor"
-"TOKandass andass"
-"TOKorass orass"
-"TOKxorass xorass"
+"TOKandass andAss"
+"TOKorass orAss"
+"TOKxorass xorAss"
 "TOKassign assign"
 "TOKtilde tilde"
-"TOKplusplus plusplus"
-"TOKminusminus minusminus"
+"TOKplusplus plusPlus"
+"TOKminusminus minusMinus"
 "TOKconstruct construct"
 "TOKblit blit"
 "TOKdot dot"
 "TOKarrow arrow"
 "TOKcomma comma"
 "TOKquestion question"
-"TOKandand andand"
-"TOKoror oror"
-"TOKpreplusplus preplusplus"
-"TOKpreminusminus preminusminus"
+"TOKandand andAnd"
+"TOKoror orOr"
+"TOKpreplusplus prePlusPlus"
+"TOKpreminusminus preMinusMinus"
 "TOKint32v int32v"
 "TOKuns32v uns32v"
 "TOKint64v int64v"
@@ -215,14 +215,14 @@
 "TOKshared shared_"
 "TOKat at"
 "TOKpow pow"
-"TOKpowass powass"
-"TOKgoesto goesto"
+"TOKpowass powAss"
+"TOKgoesto goesTo"
 "TOKvector vector"
 "TOKpound pound"
 "TOKinterval interval"
-"TOKvoidexp voidexp"
-"TOKcantexp cantexp"
-"TOKMAX MAX"
+"TOKvoidexp voidExp"
+"TOKcantexp cantExp"
+"TOKMAX max_"
 "TOKwild inout_"
 "TOKnothrow nothrow_"
 "TOKcatch catch_"
```

Some are just so over-abbreviated I don't even want to mess with them (e.g. `shl`, `lt`, etc...)  Also, I don't want to overdo it with this style stuff.

**Review by jacob-carlborg — APPROVED** [review]

**Comment by jacob-carlborg** [comment]
As mentioned in other PRs, I voted for a nested enum, `Token.Kind` or `Token.Tag`.

**Comment by JinShil** [comment]
> I voted for a nested enum, `Token.Kind`

I think that worked out well for `Prot`ection.  I'll go along with that.

**Comment by wilzbach** [comment]
> Some are just so over-abbreviated I don't even want to mess with them (e.g. shl, lt, etc...) Also, I don't want to overdo it with this style stuff.

Done. About the short ones. Well, how about:

```diff
diff --git a/src/dmd/tokens.d b/src/dmd/tokens.d
index 660c8a692..c2425a529 100644
--- a/src/dmd/tokens.d
+++ b/src/dmd/tokens.d
@@ -83,10 +83,10 @@ enum TOK : int
 
     // 54
     // Operators
-    lt,
-    gt,
-    le,
-    ge,
+    lessThan,
+    greaterThan,
+    lessOrEqual,
+    greaterOrEqual,
     equal,
     notEqual,
     identity,
@@ -107,32 +107,32 @@ enum TOK : int
     ue,
 
     // 72
-    shl,
-    shr,
-    shlAss,
-    shrAss,
-    ushr,
-    ushrAss,
+    leftShift,
+    rightShift,
+    leftShiftAssign,
+    rightShiftAssign,
+    unsignedRightShift,
+    unsignedRightShiftAssign,
     cat,
-    catAss, // ~=
-    catelemAss,
-    catdcharAss,
+    catAssgn, // ~=
+    catElemAssign,
+    catDcharAssgn,
     add,
     min,
-    addAss,
-    minAss,
+    addAssign,
+    minAssign,
     mul,
     div,
     mod,
-    mulAss,
-    divAss,
-    modAss,
+    mulAssign,
+    divAssign,
+    modAssign,
     and,
     or,
     xor,
-    andAss,
-    orAss,
-    xorAss,
+    andAssign,
+    orAssign,
+    xorAssign,
     assign,
     not,
     tilde,
```

I'm not sure about the NCEG operators - they have been [deprecated](https://dlang.org/deprecate.html#Floating%20point%20NCEG%20operators) for a long time anyways and we should rather remove them completely.

**Comment by wilzbach** [comment]
Meh testing `extern(System)` in the default `json.d` turns out to be a stupid idea:

```
<     "linkage" : "windows",

---

>     "linkage" : "c",

842c842

<     "linkage" : "windows",

---

>     "linkage" : "c",
```

(`extern(System)` is an alias for `extern(Windows)` on Windows and `extern(C)` elsewhere)
Anyhow auto-tester should hopefully be happy now.

**Comment by JinShil** [comment]
> About the short ones. Well, how about:
IMO those are all really great!.  The only reason I hesitated to change it was due to mine and @WalterBright's discussion about IDEs yesterday.  For me, since I use and IDE, I just type the first 3 or 4 characters of a name and IDE does the rest, so the length of the name is not a problem for me.  And I prefer it when the name is self-explanatory.  Just in coming up with some of my proposed names I had to search the DMD source code to figure out what they were doing.  So, IMO, your proposal is fantastic.  But that's just me.

> I'm not sure about the NCEG operators - they have been deprecated for a long time anyways and we should rather remove them completely.

I say just leave them.  It's on my list to remove them from DMD completely, so let's not waste time on them.

**Comment by JinShil** [comment]
Also if you're going to spell out "left" and "right", you might as well update `l`/`rParen` and `l`/`rBracket` too.

**Comment by jacob-carlborg** [comment]
> ```diff
> -    catAss, // ~=
> ```

Hehe 😄. That's what happens when names are shortened.

**Comment by jacob-carlborg** [comment]
> Done. About the short ones. Well, how about:

* `catAssgn` -> `catAssign`
* `catDcharAssgn` -> `catDcharAssign`

Shouldn't `cat` be `concatenate` or `concatenation`.

**Comment by jacob-carlborg** [comment]
I'm not sure how far we should take this but here are a few more shortened names:

```d
leftParen
rightParen
neg
eof
symOff
var
dotVar
dotTI
dotTD
newAnonClass
argTypes
funcString
prettyFunc
voidExp
cantExp

mul
div
mod
pow
mulAssign
divAssign
modAssign
powAssign
```

```d
int32v 
xstring
```

In the above snippet, what is the prefixed `x` and the trailing `v`?

**Comment by JinShil** [comment]
`xstring` - It looks like it's used for something called "Hexadecimal strings"  (e.g. `x"0A ae 34FE BD"`), something I didn't know existed in the D language until just now.

**Comment by JinShil** [comment]
I think `int32v` is for numeric literals, while `int32` is for "int".

**Comment by jacob-carlborg** [comment]
I think the fact that I needed to ask shows that the name is not good.

**Comment by JinShil** [comment]
OK, here's my attempt at whole-hog spell it out.  If this were my compiler I'd approve it.

```diff
@@ -1,15 +1,15 @@
 "TOKreserved reserved"
-"TOKlparen leftParen"
-"TOKrparen rightParen"
+"TOKlparen leftParentheses"
+"TOKrparen rightParentheses"
 "TOKlbracket leftBracket"
 "TOKrbracket rightBracket"
 "TOKlcurly leftCurly"
 "TOKrcurly rightCurly"
 "TOKcolon colon"
-"TOKneg neg"
-"TOKsemicolon semiColon"
+"TOKneg negate"
+"TOKsemicolon semicolon"
 "TOKdotdotdot dotDotDot"
-"TOKeof eof"
+"TOKeof endOfFile"
 "TOKcast cast_"
 "TOKnull null_"
 "TOKassert assert_"
@@ -19,32 +19,32 @@
 "TOKaddress address"
 "TOKdelete delete_"
 "TOKstar star"
-"TOKsymoff symOff"
-"TOKvar var"
-"TOKdotvar dotVar"
-"TOKdotid dotID"
-"TOKdotti dotTI"
+"TOKsymoff symbolOffset"
+"TOKvar variable"
+"TOKdotvar dotVariable"
+"TOKdotid dotIdentifier"
+"TOKdotti dotTemplateInstance"
 "TOKdottype dotType"
 "TOKslice slice"
 "TOKarraylength arrayLength"
 "TOKversion version_"
 "TOKdollar dollar"
 "TOKtemplate template_"
-"TOKdottd dotTD"
+"TOKdottd dotTemplateDeclaration"
 "TOKdeclaration declaration"
 "TOKpragma pragma_"
 "TOKdsymbol dSymbol"
 "TOKuadd uadd"
 "TOKremove remove"
-"TOKnewanonclass newAnonClass"
+"TOKnewanonclass newAnonymousClass"
 "TOKcomment comment"
 "TOKarrayliteral arrayLiteral"
 "TOKassocarrayliteral assocArrayLiteral"
 "TOKstructliteral structLiteral"
 "TOKclassreference classReference"
 "TOKthrownexception thrownException"
-"TOKdelegateptr delegatePtr"
-"TOKdelegatefuncptr delegateFuncptr"
+"TOKdelegateptr delegatePointer"
+"TOKdelegatefuncptr delegateFunctionPointer"
 "TOKlt lessThan"
 "TOKgt greaterThan"
 "TOKge greaterOrEqual"
@@ -66,13 +66,13 @@
 "TOKshrass rightShiftAssign"
 "TOKushrass unsignedRightShiftAssign"
 "TOKcatass concatenateAssign"
-"TOKcatelemass concatenateElemAssign"
+"TOKcatelemass concatenateElementAssign"
 "TOKcatdcharass concatenateDcharAssign"
 "TOKaddass addAssign"
-"TOKminass minAssign"
-"TOKmulass mulAssign"
-"TOKdivass divAssign"
-"TOKmodass modAssign"
+"TOKminass minusAssign"
+"TOKmulass multiplyAssign"
+"TOKdivass divideAssign"
+"TOKmodass modulusAssign"
 "TOKandass andAssign"
 "TOKorass orAssign"
 "TOKxorass xorAssign"
@@ -89,24 +89,24 @@
 "TOKoror orOr"
 "TOKpreplusplus prePlusPlus"
 "TOKpreminusminus preMinusMinus"
-"TOKint32v int32v"
-"TOKuns32v uns32v"
-"TOKint64v int64v"
-"TOKuns64v uns64v"
-"TOKint128v int128v"
-"TOKuns128v uns128v"
-"TOKfloat32v float32v"
-"TOKfloat64v float64v"
-"TOKfloat80v float80v"
-"TOKimaginary32v imaginary32v"
-"TOKimaginary64v imaginary64v"
-"TOKimaginary80v imaginary80v"
-"TOKcharv charv"
-"TOKwcharv wcharv"
-"TOKdcharv dcharv"
+"TOKint32v int32Literal"
+"TOKuns32v uns32Literal"
+"TOKint64v int64Literal"
+"TOKuns64v uns64Literal"
+"TOKint128v int128Literal"
+"TOKuns128v uns128Literal"
+"TOKfloat32v float32Literal"
+"TOKfloat64v float64Literal"
+"TOKfloat80v float80Literal"
+"TOKimaginary32v imaginary32Literal"
+"TOKimaginary64v imaginary64Literal"
+"TOKimaginary80v imaginary80Literal"
+"TOKcharv charLiteral"
+"TOKwcharv wcharLiteral"
+"TOKdcharv dcharLiteral"
 "TOKidentifier identifier"
 "TOKstring string_"
-"TOKxstring xstring"
+"TOKxstring hexadecimalString"
 "TOKthis this_"
 "TOKsuper super_"
 "TOKhalt halt"
@@ -186,7 +186,7 @@
 "TOKon_scope_success onScopeSuccess"
 "TOKinvariant invariant_"
 "TOKunittest unittest_"
-"TOKargTypes argTypes"
+"TOKargTypes argumentTypes"
 "TOKref ref_"
 "TOKmacro macro_"
 "TOKparameters parameters"
@@ -197,11 +197,11 @@
 "TOKline line"
 "TOKfilefullpath fileFullPath"
 "TOKmodulestring moduleString"
-"TOKfuncstring funcString"
-"TOKprettyfunc prettyFunc"
+"TOKfuncstring functionString"
+"TOKprettyfunc prettyFunction"
 "TOKshared shared_"
 "TOKat at"
-"TOKpowass powAssign"
+"TOKpowass powerAssign"
 "TOKgoesto goesTo"
 "TOKvector vector"
 "TOKpound pound"
@@ -225,15 +225,15 @@
 "TOKvoid void_"
 "TOKmodule module_"
 "TOKnew new_"
-"TOKmod mod"
+"TOKmod modulus"
 "TOKarray array"
 "TOKfile file"
-"TOKpow pow"
+"TOKpow power"
 "TOKor or"
 "TOKxor xor"
-"TOKmin min"
-"TOKmul mul"
-"TOKdiv div"
+"TOKmin minus"
+"TOKmul multiply"
+"TOKdiv divide"
 "TOKand and"
 "TOKshl leftShift"
 "TOKdot dot"
```

File: src/dmd/tokens.d [hunk 4]
```diff
@@ -24,539 +24,292 @@ import dmd.utf;
...
+    arrayLength,
+    version_,
+    module_,
+    dollar,
+    template_,
+    dotTemplateDeclaration,
+    declaration,
+    typeof_,
+    pragma_,
+    dSymbol,
```
Comment (jacob-carlborg) [inline src/dmd/tokens.d]: Is there a reason this can't be called just `symbol`? Is it to match the `Dsymbol` class?

File: src/dmd/tokens.d [hunk 5]
```diff
@@ -24,539 +24,292 @@ import dmd.utf;
...
+    vector,
+    pound,
 
     // 239
-    TOKinterval,
-    TOKvoidexp,
-    TOKcantexp,
+    interval,
+    voidExp,
+    cantExp,
```
Comment (jacob-carlborg) [inline src/dmd/tokens.d]: `Exp` -> `Expression`.

**Review by jacob-carlborg — CHANGES_REQUESTED** [review]

File: src/dmd/tokens.d (hunk 4 above)
Comment (JinShil) [reply src/dmd/tokens.d]: > Is it to match the Dsymbol class?

Yes, that was my intention.  I don't have any good reason for it not to be `symbol`.

File: src/dmd/tokens.d (hunk 5 above)
Comment (JinShil) [reply src/dmd/tokens.d]: Ok with me

File: src/dmd/tokens.d [hunk 6]
```diff
@@ -586,258 +339,258 @@ extern (C++) struct Token
...
+        TOK.extern_: "extern",
+        TOK.private_: "private",
+        TOK.package_: "package",
+        TOK.protected_: "protected",
+        TOK.public_: "public",
+        TOK.export_: "export",
+        TOK.invariant_: "invariant",
+        TOK.unittest_: "unittest",
+        TOK.version_: "version",
+        TOK.argumentTypes: "__argTypes",
```
Comment (ibuclaw) [inline src/dmd/tokens.d]: Maybe keep it as argTypes so it matches the keyword? Not bothered either way.

**Comment by ibuclaw** [comment]
> I say just leave them. It's on my list to remove them from DMD completely, so let's not waste time on them.

They're still useful for the error they produce for now.

**Comment by JinShil** [comment]
>  They're still useful for the error they produce for now.

I mean the deprecation and error periods have expired.  This feature can now be complete removed from the language, error messages and all.  So, I don't want to spend time trying to find a good name for them.

File: src/dmd/tokens.d (hunk 5 above)
Comment (wilzbach) [reply src/dmd/tokens.d]: Done.

**Review by JinShil — APPROVED** [review]

**Review by jacob-carlborg — APPROVED** [review]

**Comment by jacob-carlborg** [comment]
😄.

**Pull request merged**